### PR TITLE
WT-9179 Timestamp verification for fast-delete information (try 2)

### PR DESCRIPTION
**Pull request by sauclovian-wt**
Add code to vrfy_dsk to crosscheck the timestamps in on-disk fast-delete information.

This caused failures the first time around; digging around revealed that the unpack-time obsolete timestamp cleanup wasn't being done correctly, so I've fixed that. I am not absolutely sure this is what caused the observed failures, but it's at least consistent with the report in WT-9197.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by sauclovian-wt** [comment]
Looks like the WT-8929 code (the new paired typo checker) is having some issues :-(

File: src/btree/bt_vrfy_dsk.c [hunk 1]
```diff
@@ -237,6 +237,89 @@ __verify_dsk_value_validity(WT_SESSION_IMPL *session, WT_CELL_UNPACK_KV *unpack,
...
+          cell_num - 1, tag);
+
+    /*
+     * The other elements of the structure are not stored on disk and are set unconditionally by the
+     * unpack code, so just assert about them. Prepared fast-truncates are not allowed to be
+     * evicted.
+     */
+    WT_ASSERT(session, unpack->page_del.prepare_state == 0);
+    WT_ASSERT(session, unpack->page_del.previous_ref_state == WT_REF_DISK);
+    WT_ASSERT(session, unpack->page_del.committed == true);
```
Comment (keithbostic) [inline src/btree/bt_vrfy_dsk.c:318]: Verify is also supposed to detect corruption, so these shouldn't be asserts, they should be "normal" failures.

**Review by keithbostic — APPROVED** [review]
lgtm

File: src/btree/bt_vrfy_dsk.c (hunk 1 above)
Comment (sauclovian-wt) [reply src/btree/bt_vrfy_dsk.c:318]: As per the comment, those aren't read from disk, so asserts seemed like the right thing...

Comment (keithbostic) [reply src/btree/bt_vrfy_dsk.c:318]: Got it, makes sense.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**